project/activities/views.py

```diff
@@ -7,6 +7,73 @@
 
 from .forms import ActivityModelForm
 from .models import Activity
+from .serializers import ActivitySerializer
+
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework import status
+
+
+@api_view(['GET', 'POST'])
+def activity_list(request):
+    """
+    Circle member(s) can create a Circle Activity via a POST request
+    Circle members and coordinators can view all Circle Activities via a GET request
+    """
+    circle_id = Activity.objects.values_list("circle_id", flat=True).first()
+    circle = Circle.objects.get(id=circle_id)
+
+    if request.user in circle.companions:
```

Comment:
When using class-based views, we can use `permission_classes` for re-usable permission logic:

https://www.django-rest-framework.org/tutorial/4-authentication-and-permissions/#object-level-permissions